src/middleware.ts

```diff
@@ -0,0 +1,36 @@
+import type { RequestContext, ResponseContext, ConfigurationOptions, Middleware } from './gen/index.ts';
+import { PromiseMiddleware } from './gen/middleware.js';
+
+//WithHeaderMiddleware returns Middleware[] that sets a header value
+export function WithHeaderMiddleware(key: string, value: string): Middleware[] {
```

Comment:
The UpperCamelCase function names make this feel more like Golang than JavaScript.